plans component

kwiedenhaupt opened the pull request:
Purpose of the PR:
To create a plans page that contains cards that outline plans that students can sign up for

Desktop View:
![image](https://user-images.githubusercontent.com/111075855/230806075-c7b7e4f1-4cef-4dbd-8115-4f410fc5b6de.png)

Mobile View:
![image](https://user-images.githubusercontent.com/111075855/230806117-deff2340-e7c2-4fa9-991f-608bcd533e1e.png)
![image](https://user-images.githubusercontent.com/111075855/230806147-ea145e0d-502d-4946-a047-f9e23ae03038.png)
![image](https://user-images.githubusercontent.com/111075855/230806168-36ff3b51-391f-471e-accb-72bfcb3a98a8.png)

Structure of the Plans Page
1. Index page that houses the PlansCard. The data is contained in an array of cards so that it can easily be expanded or retracted.
2. PlansCard contains a PlansRow component. Again, the data is contained in an array that is mapped out.
3. PlansRow component has the redX and greenCheck images taken from the Figma sketches and then a description next to it.  It is set up as a grid so that the description is able to populate the width of the card.

TO FOLLOW IN ANOTHER PR:
1. Populate the Data
2. Create the clickable buttons that contain the paths that we want to send them to with handleClick functions that are drilled down.

iamnotreddy commented on pages/studentPortal/plans/index.tsx:
```diff
@@ -0,0 +1,54 @@
+      description: "desc 1",
+      price: "$0",
+      buttonLabel: "Sign Up",
+      onClick: () => console.log("clicked"),
```
remove the console.logs

iamnotreddy commented on pages/studentPortal/plans/index.tsx:
```diff
@@ -0,0 +1,54 @@
+
+  return (
+    <div>
+      <div>
```
extra div

iamnotreddy reviewed (changes requested):
things to address:
- fix unnecessary spacing both on the main page and inside the card
- add a spacing on the left / right so the row PlanRow doesn't touch edge of the card
- let's try to get this render without a scroll-bar, maybe take a look at overflow in the tailwind docs
-

kwiedenhaupt commented:
CHANGES MADE: 
1. changed the rendering to getServerSideProps
2. re-structured the data so that the planRowData[] is included in the plansCardData[].
3. adjusted spacing of the cards
4. added content and links to the onClick properties of the buttons
5. changed the spacing so that no matter if there are varying numbers of columns the cards will start at the same height
6. added the ....and more! conditional optionalMessage

FURTHER CONCERN:
it seems like I am still having the overflow problem? I tried throwing the overflow-hidden class into the parent container of the plans page and it didn't seem to help...same with the plansCard.  There is no fixed height to the plans card that would be inadvertently causing it to be longer than I expected.  not sure.

kwiedenhaupt commented:
![image](https://user-images.githubusercontent.com/111075855/231168608-a8a72231-dae9-441b-be54-c93c535ff1fd.png)

iamnotreddy commented on pages/plans.tsx:
```diff
@@ -0,0 +1,131 @@
+export const plansCardData = [
```
let's declare this inside the getServerSieProps function instead of at the top of the file

iamnotreddy reviewed (approved):
couple small changes but nice work!

vithushan19 reviewed (approved):
lgtm

kwiedenhaupt commented on pages/plans.tsx:
```diff
@@ -0,0 +1,131 @@
+    price: "Custom",
+    buttonLabel: "Book a Call",
+    onClick: "https://joinskillify.com/call",
+    optionalMessage: "...and more!",
```
done!

kwiedenhaupt commented:
Updated UI with the changes:
![image](https://user-images.githubusercontent.com/111075855/232820118-20754818-eddd-4dfc-b100-9849ffbe703b.png)
